### DISCUSSION TITLE
Integration of an own Data Plane Client

### DISCUSSION THREAD

**Discussion by tnolden** [eclipse-edc/Connector#3164] [Q&A]
I wrote my own client extension as part of my bachelor thesis. At which point can I assign that the Connector uses this client / extension for websites of a certain URL or type?

**Comment by jimmarino** [comment]
Sorry, but your question is difficult to follow. Can you please rephrase it?

**Reply by tnolden** [reply]
Of course! As part of my bachelor thesis, I had to integrate an existing web service as a data source / data sink into a dataspace with the EDC Connector. Since the webservice has some peculiarities regarding authentication, especially in the local test environment, I had to write a custom client to connect to it. I wrote a custom extension for the webservice, which is based on the DataPlaneHttp extension and passes my client instead of the usual client. The extension works so far. However, I don't know where to assign the connector to use my extension for certain urls. I figured that this probably goes through the data plane selector, but so far I haven't figured out where / to what extent I can define this. 

The definition would be:
Use for all urls whith the domain example: https: //example.com/....  the extension NewClient. With the Azure extensions there was also an AzureStorage type for example. Therefore I had also considered to create a new type over the type manage for the kind Url.  I could use the type in the asset creation, but I also don't know, how to assign the use of the extension for a specific type.

It would be great if you can help me.

**Reply by jimmarino** [reply]
No knowing the specifics, creating a new client sounds fine. If you need to selectively use the client, you can define a special data transfer type.
